toolkit/tools/pkg/imagecustomizerlib/distrohandler.go

```diff
@@ -0,0 +1,94 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package imagecustomizerlib
+
+import (
+	"context"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/imagecustomizerapi"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safechroot"
+)
+
+// PackageManagerType represents the type of package manager
+type PackageManagerType string
+
+const (
+	packageManagerTDNF PackageManagerType = "tdnf"
+	packageManagerDNF  PackageManagerType = "dnf"
+)
+
+// PackageType represents the type of package format
+type PackageType string
+
+const (
+	packageTypeRPM PackageType = "rpm"
+	packageTypeDeb PackageType = "deb"
+)
+
+// DistroName represents the distribution name
+type DistroName string
+
+const (
+	distroNameAzureLinux DistroName = "azurelinux"
+	distroNameFedora     DistroName = "fedora"
+)
+
+// packageManagerHandler represents the interface for package manager implementation details
+type packageManagerHandler interface {
+	// Package manager configuration
+	getPackageManagerBinary() string
+	getPackageType() PackageType
+	getReleaseVersion() string
+	getConfigFile() string
+	getPackageSourceDir() string
+
+	// Package manager specific output handling
+	createOutputCallback() func(string)
+}
+
+// rpmPackageManagerHandler represents the interface for RPM-based package managers (TDNF, DNF)
+type rpmPackageManagerHandler interface {
+	packageManagerHandler
+
+	// RPM-specific functionality is already covered by packageManagerHandler
+	// This interface exists to enforce type safety for RPM-only operations
+}
+
+// distroHandler represents the interface for distribution-specific configuration
+type distroHandler interface {
+	// Distribution identification
+	getDistroName() DistroName
+
+	// Get the package manager for this distribution
+	getPackageManager() packageManagerHandler
+
+	// Package management operations
+	managePackages(ctx context.Context, buildDir string, baseConfigPath string, config *imagecustomizerapi.OS,
+		imageChroot *safechroot.Chroot, toolsChroot *safechroot.Chroot,
+		rpmsSources []string, useBaseImageRpmRepos bool, snapshotTime string) error
+}
+
+// NewDistroHandler creates the appropriate distro handler with version support
+func NewDistroHandler(distroName string, version string) distroHandler {
```

Comment:
Please key everything off `targetos.TargetOs`. Don't go passing around raw distro names and distro version strings.

By the way, feel free to extend `TargetOs` as needed. For example, you could add a `GetDistro` and `GetDistroVersion` functions to `TargetOs`.